validation/src/test/java/dk/kvalitetsit/itukt/validation/service/ExpressionTests.java

```diff
@@ -0,0 +1,298 @@
+    final ValidationInput validationInputWithHistory =
+            new ValidationInput("", "", empty(), List.of(), inputAge, 0, inputIndication, Optional.of(existingMedications));
+
+    static void assertErrorMessage(String v, Optional<ValidationError> o) {
+        assertEquals(Optional.of(v), o.map(ValidationError::errorMessage));
+    }
+
+    static Stream<Arguments> allValidNumberConditionCombinations() {
+        return Stream.of(Operator.values())
+                .flatMap(ExpressionTests::toValidAgeConditionExpression)
+                .map(Arguments::of);
+    }
+
+    static Stream<AgeConditionExpression> toValidAgeConditionExpression(Operator op) {
+        var validatingAges = switch (op) {
+            case EQUAL -> Stream.of(inputAge);
+            case LESS_THAN_OR_EQUAL_TO -> Stream.of(inputAge, inputAge + 1);
+            case GREATER_THAN_OR_EQUAL_TO -> Stream.of(inputAge, inputAge - 1);
+            case GREATER_THAN -> Stream.of(inputAge - 1);
+            case LESS_THAN -> Stream.of(inputAge + 1);
+        };
+        return validatingAges.map(v -> new AgeConditionExpression(op, v));
+    }
+
+    static Stream<Arguments> allErrorNumberConditionCombinations() {
+        return Stream.of(Operator.values())
+                .map(ExpressionTests::toErrorAgeConditionExpression)
+                .map(Arguments::of);
+    }
+
+    static ExpressionAndExpectedError toErrorAgeConditionExpression(Operator op) {
+        var failingAgesWithError = switch (op) {
+            case EQUAL -> new ExpectedAgeAndError(inputAge + 1, "alder skal være " + (inputAge + 1));
+            case GREATER_THAN -> new ExpectedAgeAndError(inputAge + 1, "alder skal være større end " + (inputAge + 1));
+            case LESS_THAN_OR_EQUAL_TO -> new ExpectedAgeAndError(inputAge - 1, "alder skal være mindre end eller lig " + (inputAge - 1));
+            case GREATER_THAN_OR_EQUAL_TO -> new ExpectedAgeAndError(inputAge + 1, "alder skal være større end eller lig " + (inputAge + 1));
+            case LESS_THAN -> new ExpectedAgeAndError(inputAge - 1, "alder skal være mindre end " + (inputAge - 1));
+        };
+        return new ExpressionAndExpectedError(
+                new AgeConditionExpression(op, failingAgesWithError.age),
+                failingAgesWithError.error);
+    }
+
+    record ExpectedAgeAndError(Integer age, String error) {}
+
+    record ExpressionAndExpectedError(Expression expression, String expectedError) {}
+
+    @ParameterizedTest
+    @MethodSource("allValidNumberConditionCombinations")
+    void validateNumber_noError(AgeConditionExpression expression) {
+        assertEquals(empty(), expression.validates(validationInput));
+    }
+
+
+    @ParameterizedTest
+    @MethodSource("allErrorNumberConditionCombinations")
+    void validateNumber_error(ExpressionAndExpectedError expressionAndError) {
+        assertErrorMessage(expressionAndError.expectedError, expressionAndError.expression.validates(validationInput));
+    }
```

Comment:
```suggestion
    @Test
    void validateNumber_noError() {
        Stream.of(Operator.values())
                .flatMap(ExpressionTests::toValidAgeConditionExpression)
                .forEach(x -> assertEquals(empty(), x.validates(validationInput)));
    }
```


Jeg vil fortrække at gøre tests "self-contained"

Reply:
Jeg tænkte også på at gøre det. Problemet er at testen stopper ved den første assert der fejler, så du kan ende med at skulle gøre testene af flere omgange hvis mere end én assert fejler her.

Reply:
Som aftalt har jeg refactored dem ud i individuelle tests.